library/core/src/mem/mod.rs

```diff
@@ -627,12 +627,22 @@ pub unsafe fn zeroed<T>() -> T {
     }
 }
 
-/// Bypasses Rust's normal memory-initialization checks by pretending to
-/// produce a value of type `T`, while doing nothing at all.
-///
-/// **This function is deprecated.** Use [`MaybeUninit<T>`] instead.
-///
-/// The reason for deprecation is that the function basically cannot be used
+/// Produces a value of type `T` that may be of any imaginable
+/// bit representation in memory.
+/// The exact value that is produced by this function is subject to no
+/// guarantees whatsoever, and cannot be relied upon.
+/// The value returned by this function may differ between implementations,
+/// between different versions of the same implementation,
+/// between different compilations of the exact same code with the exact same compiler,
+/// between different executions of the same program,
+/// between different invocations within a single program,
+/// between different *uses* of the same returned value,
+/// or for any other reason whatsoever, with no warning.
+///
+/// **This function is deprecated with extreme prejudice.**
+/// It is replaced by [`MaybeUninit<T>`], which must be used instead of this function.
+///
+/// The reason for deprecation is that this function cannot be used
```

Comment:
You removed the "basically", which makes this statement technically wrong. E.g. this code is fine:
```rust
let x: [MaybeUninit<i32>; 32] = mem::uninitialized();
```

Reply:
You caught me, I was trying to make the language scarier. :P